### PR TITLE
[+] Implement a new dialog for adding an entry

### PR DESCRIPTION
**Pull request by Yunshan-522**
I have designed and implemented a new dialog to add an entry. However, after I enter the information in this dialog, the information cannot be added to the bill.

### CODE REVIEW

File: src/main/java/billgates/view/gui/MainFrame.java [hunk 1]
```diff
@@ -84,4 +86,12 @@ public UserJoinController getUserJoinController() {
     public void setUserJoinController(UserJoinController userJoinController) {
         this.userJoinController = userJoinController;
     }
+
+    public InsertEntryController getInsertEntryController() {
+        return insertEntryController;
```
Comment (ScottCTD) [inline src/main/java/billgates/view/gui/MainFrame.java:91]: Convention: Add `this.` before using an instance variable.

File: src/main/java/billgates/view/gui/ActionPanel.java [hunk 2]
```diff
@@ -215,7 +217,48 @@ private void backFromSplit() {
...
+        myPanel.add(new JLabel("to:"));
+        myPanel.add(toField);
+        myPanel.add(new JLabel("location:"));
+        myPanel.add(locationField);
+
+        int result = JOptionPane.showConfirmDialog(null, myPanel,
+                "Information about the new entry", JOptionPane.OK_CANCEL_OPTION);
+        if (result == JOptionPane.OK_OPTION) {
+            String date1 = dateField.getText();
+            ZonedDateTime date = ZonedDateTime.parse(date1);
```
Comment (ScottCTD) [inline src/main/java/billgates/view/gui/ActionPanel.java:249]: This cases some exceptions with formatting datetime.

The solution could be replacing this line with 
```java
LocalDateTime localDateTime = LocalDateTime.parse(date1, DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
ZonedDateTime date = ZonedDateTime.of(localDateTime, ZoneId.systemDefault());
```

**Review by ScottCTD — CHANGES_REQUESTED** [review]
After fixing the bug I mentioned, this could be a working functionality. GOOD!

Future TODO:

- [ ] [Enhancement] Use some datetime picker to edit the date.
- [ ] [Enhancement] Add support for adding a splitter entry.
- [ ] [Improvement] Create a separate class for the dialog of adding an entry.

File: src/main/java/billgates/view/gui/ActionPanel.java [hunk 3]
```diff
@@ -215,7 +217,48 @@ private void backFromSplit() {
...
 
     private void addEntry() {
-        // TODO: Call the controller of InsertEntryUseCase
+        JTextField dateField = new JTextField(20);
+        JTextField valueField = new JTextField(20);
+        JTextField currencyField = new JTextField(20);
+        JTextField descriptionField = new JTextField(20);
+        JTextField fromField = new JTextField(20);
+        JTextField toField = new JTextField(20);
+        JTextField locationField = new JTextField(20);
```
Comment (Lei-Tin) [inline src/main/java/billgates/view/gui/ActionPanel.java:226]: Some of these fields needs to align with the information that we have setup in the database. 

For reference: 
image

Value should have length of 16, currency should have length of 3. 

All the rest might go over 20 characters, but I'm not really sure if we can specify them to be unlimited when initializing?

**Review by Lei-Tin — CHANGES_REQUESTED** [review]
Looking pretty good, just some small changes to be fixed for me.

File: src/main/java/billgates/view/gui/ActionPanel.java (hunk 3 above)
Comment (ScottCTD) [reply src/main/java/billgates/view/gui/ActionPanel.java:226]: GUI stuff doesn't affect database stuff.

`new JTextField(20)` is kind of like the length of the input field is approximately 20 characters.

**Review by ScottCTD — DISMISSED** [review]
Currently GREAT!
We have some future TODOs:

- [ ][Enhancement] Use some datetime picker to edit the date.
- [ ][Enhancement] Add support for adding a splitter entry.
- [ ][Improvement] Create a separate class for the dialog of adding an entry.

**Pull request merged**